python/pyarrow/scalar.pxi

```diff
@@ -793,7 +793,7 @@ cdef class MapScalar(ListScalar):
         """
         arr = self.values
         if array is None:
```

Comment:
Not related to this PR, but just noticing: this line uses `array` variable, but that is not defined in this scope (only `arr` is), so this `array` variable is actually referring to the `pyarrow.lib.array(..)` function ..

Reply:
I think so too now. :-)
I've renamed it to `ary`.